internal/pkg/service/stream/storage/test/testcase/testcase.go

```diff
@@ -55,26 +55,32 @@ func (tc *WriterTestCase) Run(t *testing.T) {
 
 	d, mock := dependencies.NewMockedLocalStorageScope(t)
 	cfg := mock.TestConfig()
+	cfg.Storage.Level.Local.Writer.WatchDrainFile = false
 
 	// Start statistics collector
-	writerEvents := events.New[diskwriter.Writer]()
-	collector.Start(d, writerEvents, cfg.Storage.Statistics.Collector, cfg.NodeID)
+	pipelineEvents := events.New[encoding.Pipeline]()
+	collector.Start(d, pipelineEvents, cfg.Storage.Statistics.Collector, cfg.NodeID)
 
 	// Open volume
-	opts := []writerVolume.Option{writerVolume.WithWatchDrainFile(false)}
 	now := d.Clock().Now()
 	volPath := t.TempDir()
 	spec := volume.Spec{NodeID: "my-node", Path: volPath, Type: "hdd", Label: "1"}
-	vol, err := writerVolume.Open(ctx, d.Logger(), d.Clock(), writerEvents, cfg.Storage.Level.Local.Writer, spec, opts...)
+	vol, err := diskwriter.Open(ctx, d.Logger(), d.Clock(), cfg.Storage.Level.Local.Writer, spec, events.New[diskwriter.Writer]())
 	require.NoError(t, err)
 
 	// Create a test slice
 	slice := tc.newSlice(t, vol)
 	filePath := filepath.Join(volPath, slice.LocalStorage.Dir, slice.LocalStorage.Filename)
 
-	// Create writer
-	w, err := vol.OpenWriter(slice)
-	require.NoError(t, err)
+	// Create encoder pipeline
+	openPipeline := func() encoding.Pipeline {
+		diskWriter, err := vol.OpenWriter(slice)
+		require.NoError(t, err)
+		pipeline, err := encoding.NewPipeline(ctx, d.Logger(), d.Clock(), cfg.Storage.Level.Local.Encoding, slice, diskWriter, pipelineEvents)
+		require.NoError(t, err)
+		return pipeline
+	}
+	writer := openPipeline()
 
```

Comment:
:exclamation: Here is nice example of changes:
- Instead of the one writer, here is `diskwriter.Writer` (Writer Node) and `encoding.Pipeline` (Source Node).
- For now, they are connected directly, network connection will be in the next PR.